lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java

```diff
@@ -1142,33 +1195,92 @@ private static Object preprocessWriteValueHive(Type type, Object value)
     static RecordWriter createOrcRecordWriter(File outputFile, Format format, CompressionKind compression, Type type)
             throws IOException
     {
-        JobConf jobConf = new JobConf(newEmptyConfiguration());
-        OrcConf.WRITE_FORMAT.setString(jobConf, format == ORC_12 ? "0.12" : "0.11");
-        OrcConf.COMPRESS.setString(jobConf, compression.name());
-
-        return new OrcOutputFormat().getHiveRecordWriter(
-                jobConf,
-                new Path(outputFile.toURI()),
-                Text.class,
-                compression != NONE,
-                createTableProperties("test", getJavaObjectInspector(type).getTypeName()),
-                () -> {});
+        StandardStructObjectInspector objectInspector = getStandardStructObjectInspector(ImmutableList.of("test"), ImmutableList.of(getJavaObjectInspector(type)));
+
+        return RecordWriterBuilder.builder(outputFile, format, compression)
+                .putColumns(objectInspector)
+                .withBloomFilter(objectInspector, 0.50)
+                .build();
+    }
+
+    private static RecordWriter createOrcRecordWriter(File outputFile, Format format, CompressionKind compression, StandardStructObjectInspector objectInspector)
+            throws IOException
+    {
+        return RecordWriterBuilder.builder(outputFile, format, compression)
+                .putColumns(objectInspector)
+                .withStripeSize(120000L)
+                .build();
     }
 
-    static SettableStructObjectInspector createSettableStructObjectInspector(String name, Type type)
+    private static class RecordWriterBuilder
     {
-        return getStandardStructObjectInspector(ImmutableList.of(name), ImmutableList.of(getJavaObjectInspector(type)));
+        private final JobConf jobConf;
+        private final File file;
+
+        private Class<? extends Writable> valueClass = Text.class;
+        private final CompressionKind compression;
+        private Progressable reporter = () -> {};
+        private Properties tableProperties = new Properties();
+
+        private RecordWriterBuilder(File file, Format format, CompressionKind compression)
+        {
+            this.jobConf = new JobConf(newEmptyConfiguration());
+            this.file = file;
+            this.compression = compression;
+            OrcConf.WRITE_FORMAT.setString(jobConf, format == ORC_12 ? "0.12" : "0.11");
+            OrcConf.COMPRESS.setString(jobConf, compression.name());
+        }
+
+        public static RecordWriterBuilder builder(File file, Format format, CompressionKind compression)
+        {
+            return new RecordWriterBuilder(file, format, compression);
+        }
+
+        public RecordWriterBuilder putColumns(StandardStructObjectInspector objectInspector)
```

Comment:
`putColumns` sounds like adding columns, but this logic overwrites existing values if the method is called more than once. How about renaming to `withColumns` or moving to `builder` arguments?

Reply:
make sense. changed to `withColumns`.